crates/rspack_core/src/init_fragment.rs

```diff
@@ -29,4 +66,79 @@ pub enum InitFragmentStage {
   StageAsyncHarmonyImports,
 }
 
-pub type ChunkInitFragments = HashMap<String, InitFragment>;
+pub type ChunkInitFragments = HashMap<String, NormalInitFragment>;
+
+pub fn render_init_fragments(
+  source: BoxSource,
+  fragments: &mut Vec<&mut NormalInitFragment>,
+) -> BoxSource {
+  // here use sort_by_key because need keep order equal stage fragments
+  fragments.sort_by_key(|m| m.stage);
+  // merge same init fragments
+  fragments.dedup();
+
+  let mut sources = ConcatSource::default();
+
+  fragments.iter_mut().for_each(|f| {
+    sources.add(RawSource::from(std::mem::take(&mut f.content)));
+  });
+
+  sources.add(source);
+
+  fragments.iter_mut().rev().for_each(|f| {
```

Comment:
Just out of suggestion, can we implement a `Fromiterator` trait on the `rspack_source` side for making upstream code base a little bit cleaner?

Reply:
`InitFragment` is rspack designed struct, it is un-relevant to `rspack_source`, so i think put it into `rspack` is good for me.